### ISSUE TITLE
Path validation error while trying to cache gradle

### ISSUE THREAD

**Issue by biswanathTewari** [actions/setup-java#528]
**Description:**
Path validation error while trying to cache gradle

**Task version:**
v3

**Platform:**
- [ ] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [ ] Hosted
- [ ] Self-hosted

**Repro steps:**  
Setup java using actions/setup-java, with the following configuration
-   distribution: 'temurin'
-   java-version: '17'
-   cache: 'gradle'

**Expected behavior:**
Cache the gradle post setting up java

**Actual behavior:**
Gradle is not cached and throws a warning: "Error: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.".

**Comment by dmitry-shibanov** [comment]
Hello @biswanathTewari. Thank you for your report. We'll take a look at the issue.

**Comment by dmitry-shibanov** [comment]
@biswanathTewari Could you please provide a public repository to reproduce the issue? I've tried to reproduce the issue but for me it works as expected.

**Comment by biswanathTewari** [comment]
@dmitry-shibanov can you please have a look at the following:- 
1. [action report](https://github.com/biswanathTewari/teslaClone/actions/runs/5950467445/job/16138315069)
2. [Repo link](https://github.com/biswanathTewari/teslaClone)

**Comment by dmitry-shibanov** [comment]
Hello @biswanathTewari. I think it is an expected behaviour because the action has nothing to restore that is why it tries to save. But the `gradle` tool is not called to install dependencies.

**Comment by biswanathTewari** [comment]
@dmitry-shibanov Yes, came to the same conclusion, if we create a build then gradle is being cached. Closing this issue. Thanks for the help.
